src/main/java/org/opentripplanner/graph_builder/module/osm/EscalatorProcessor.java

```diff
@@ -0,0 +1,71 @@
+package org.opentripplanner.graph_builder.module.osm;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.opentripplanner.graph_builder.issue.api.DataImportIssueStore;
+import org.opentripplanner.openstreetmap.model.OSMWay;
+import org.opentripplanner.street.model.edge.EscalatorEdge;
+import org.opentripplanner.street.model.vertex.IntersectionVertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains the logic for extracting elevators out of OSM data
+ */
+class EscalatorProcessor {
```

Comment:
Yes, it makes sense not to go through all the ways for the escalators separately if it can be avoided. This could probably be done in the same function that generates the street edges as well

Reply:
You could move the code that generates the StreetEdges into an own class (or classes) from the OsmModule and just initialize these processors in the OsmModule. I think you can iterate over the ways there and then just call either `buildEscalatorEdge` or `buildStreetEdge` method from one of those processor depending when is appropriate for the way.